atomicapp/cli/main.py

```diff
@@ -399,9 +411,12 @@ def run(self):
         if args.mode:
             args.action = args.mode     # Allow mode to override 'action'
         cmdline.insert(0, args.action)  # Place 'action' at front
+        # Setup logging (now with arguments from cmdline)
+        Logging.setup_logging(args.verbose, args.quiet, args.logging_output)
         logger.info("Action/Mode Selected is: %s" % args.action)
 
         # Finally, parse args and give error if necessary
+        logger.debug("Final parsed cmdline: {}".format(' '.join(cmdline)))
         args = self.parser.parse_args(cmdline)
```

Comment:
Moving it down did make the test pass. Now just need to figure out what part of my logic is flawed in the first part.